kedro/extras/datasets/networkx/gml_dataset.py

```diff
@@ -0,0 +1,162 @@
+
+from kedro.io.core import (
+    AbstractVersionedDataSet,
+    Version,
+    get_filepath_str,
+    get_protocol_and_path,
+)
+
+
+class GMLDataSet(AbstractVersionedDataSet):
+    """``GMLDataSet`` loads and saves graphs to a GML file using an
+    underlying filesystem (e.g.: local, S3, GCS). ``NetworkX`` is used to
+    create GML data.
+    See https://networkx.org/documentation/stable/tutorial.html for details.
+
+    Example:
+    ::
+
+        >>> from kedro.extras.datasets.networkx import GMLDataSet
+        >>> import networkx as nx
+        >>> graph = nx.complete_graph(100)
+        >>> graph_dataset = GMLDataSet(filepath="test.gml")
+        >>> graph_dataset.save(graph)
+        >>> reloaded = graph_dataset.load()
+        >>> assert nx.is_isomorphic(graph, reloaded)
+
+    """
+
+    DEFAULT_LOAD_ARGS = {}  # type: Dict[str, Any]
+    DEFAULT_SAVE_ARGS = {}  # type: Dict[str, Any]
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        filepath: str,
+        load_args: Dict[str, Any] = None,
+        save_args: Dict[str, Any] = None,
+        version: Version = None,
+        credentials: Dict[str, Any] = None,
+        fs_args: Dict[str, Any] = None,
+    ) -> None:
+        """Creates a new instance of ``GMLDataSet``.
+
+        Args:
+            filepath: Filepath in POSIX format to the NetworkX graph GML file.
+            load_args: Arguments passed on to ```networkx.read_gml``.
+                See the details in
+                https://networkx.org/documentation/stable/reference/readwrite/generated/networkx.readwrite.gml.read_gml.html
+            save_args: Arguments passed on to ```networkx.write_gml``.
+                See the details in
+                https://networkx.org/documentation/stable/reference/readwrite/generated/networkx.readwrite.gml.write_gml.html
+            version: If specified, should be an instance of
+                ``kedro.io.core.Version``. If its ``load`` attribute is
+                None, the latest version will be loaded. If its ``save``
+                attribute is None, save version will be autogenerated.
+            credentials: Credentials required to get access to the underlying filesystem.
+                E.g. for ``GCSFileSystem`` it should look like `{"token": None}`.
+            fs_args: Extra arguments to pass into underlying filesystem class constructor
+                (e.g. `{"project": "my-project"}` for ``GCSFileSystem``), as well as
+                to pass to the filesystem's `open` method through nested keys
+                `open_args_load` and `open_args_save`.
+                Here you can find all available arguments for `open`:
+                https://filesystem-spec.readthedocs.io/en/latest/api.html#fsspec.spec.AbstractFileSystem.open
+                All defaults are preserved, except `mode`, which is set to `r` when loading
+                and to `w` when saving.
+        """
+        _fs_args = deepcopy(fs_args) or {}
```

Comment:
```suggestion
        _fs_args = deepcopy(fs_args) or {}
        _fs_open_args_load = _fs_args.pop("open_args_load", {})
        _fs_open_args_save = _fs_args.pop("open_args_save", {})
```
To pro-process any fsspec args passed